power_grid_model_io_native_c/power_grid_model_io_native/include/power_grid_model_io_native/vnf_converter/vnf_pgm_converter.hpp

```diff
@@ -17,7 +17,7 @@ class PgmVnfConverter {
 
     // Public member functions
     void parse_vnf_file();
-    power_grid_model::ConstDataset* convert_input(power_grid_model::ConstDataset* dataset);
+    void convert_input(power_grid_model::ConstDataset* dataset);
```

Comment:
1. Why this change? If we are expecting not to return anything, the dataset provided by the user can't be touched as it is a `ConstDataset`, hence this signature doesn't make much sense. Because we either return a converted dataset (I assume this function should convert the original dataset to something else) or we ask the user to provide another dataset to be filled up with the converted data (which I don't think is the way to go). Same with others below.
2. `convert_input(power_grid_model::ConstDataset* dataset)` should probably be `convert_input(power_grid_model::ConstDataset const* dataset)` or even`convert_input(power_grid_model::ConstDataset const* const dataset)`. Please double check.

Reply:
This function should take the deserialized data that we have obtained during parse_vnf_file(), convert it to pgm format and place it into a dataset supplied by the user. We can't own the data, the user must own the data/buffer, because when the conversion process is finished we will be freeing up the memory and everything that we own will be destroyed. While the user might want to work on the data longer.

Reply:
Indeed my point. Shoulnd't then the signature reflect that either the user must pass a data buffer as an argument or the function return it? I would argue the first since is compliant with the pgm core C API.

Also, if this takes the dataset created by the deserializer, shouldn't it be a `WritableDataset`?

Reply:
We have a WritableDataset as a member value of the class, so we will be able to utilize the desiarilized data at any time after the parse_vnf_file().

Indeed we need a data buffer from the user that we could fill in, based on this https://power-grid-model.readthedocs.io/en/stable/advanced_documentation/c-api.html#dataset-views I thought that const dataset is what we need

Reply:
> We have a WritableDataset as a member value of the class, so we will be able to utilize the desiarilized data at any time after the parse_vnf_file().

Ok, I did not check that beforehand. Thanks for clarifying.

> Indeed we need a data buffer from the user that we could fill in, based on this https://power-grid-model.readthedocs.io/en/stable/advanced_documentation/c-api.html#dataset-views I thought that const dataset is what we need

I now understand what is that you meant. We should indeed create a `ConstDataset` for the user, but it should be returned, such as in the `PGM_create_dataset_const*` functions of the PGM C API. If we already take the `ConstDataset` as an argument, then we can not manipulate it because we don't own it.